onnxruntime/core/providers/dnnl/subgraph/dnnl_dequantizelinear.cc

```diff
@@ -25,6 +25,12 @@ void DnnlDequantizeLinear::CreatePrimitive(DnnlSubgraphPrimitive& sp, DnnlNode&
 
   // Check if scale and zp are scalars
   bool isScalar = sp.IsScalar(node.Input(IN_X_SCALE));
+  // Check if zp is provided and non zero
+  bool isZeroPointUseful = false;
+  if (node.Input(IN_X_ZERO_POINT).Exists()) {
+    auto mem = sp.GetMemory(node.Input(IN_X_ZERO_POINT));
+    isZeroPointUseful = isZeroPointNonZero(&mem);
+  }
```

Comment:
I think you need to add an additional check here. I think you need to check if the input is a constant initializer. This may cause runtime issues if the first call to this function the zero point is zero but the next time it is not zero.  

Insde the node capabilities code I checked for contant intilizer for the pow op as follows `graph_viewer.IsConstantInitializer(node_inputs[1]->Name(), true)`